src/Rector/AbstractRector.php

```diff
@@ -163,6 +167,14 @@ public function autowire(
         $this->docBlockUpdater = $docBlockUpdater;
     }
 
+    /**
+     * Process Node of matched type
+     * @phpstan-param TNodeType $node
+     * @phpstan-ignore-next-line Abstract method is only for proxying generics information
```

Comment:
Do you actually need to proxy those ? Aren't they inherited ? That looks like a bug (or a missing feature) in phpstan in that case.

Reply:
I had some error reported at some point, but probably it was caused by superfluous `@return` in `getNodeTypes()` implemented in the rule, that was overriding generic - I've found it after adding abstract method in the `AbstractRector` and did not check if it works without it. Now I checked it and it seems to be working, which is great, thanks 🙂